verticapy/pipeline/_transform.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+"""
+Copyright  (c)  2018-2024 Open Text  or  one  of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+
+You may obtain a copy of the License at:
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless  required  by applicable  law or  agreed to in
+    writing, software  distributed  under the  License is
+    distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+    OR CONDITIONS OF ANY KIND, either express or implied.
+    See the  License for the specific  language governing
+    permissions and limitations under the License.
+"""
+"""
+This script runs the Vertica Machine Learning Pipeline Transforming.
+"""
+
+
+from queue import Queue
+
+import verticapy as vp
+from verticapy.core.vdataframe.base import vDataFrame
+
+
+def reset_queue(column_queue: Queue) -> Queue:
+    """
+    Return a new Queue where all columns haven't been tried.
+    """
+    new_queue = Queue()
+    while not column_queue.empty():
+        col, _ = column_queue.get()
+        new_queue.put((col, False))
+    return new_queue
+
+
+def transformation(transform: dict, table: str) -> vDataFrame:
+    """
+    Run the transformation step
+    of the pipeline.
+
+    Parameters
+    ----------
+    transform: dict
+        YAML object which outlines the steps of the operation.
+    table: str
+        The name of the table the pipeline is ingesting to.
+
+    Returns
+    -------
+    vDataFrame
+        The transformed vDataFrame.
+    """
+    vdf = vp.vDataFrame(table)
+
+    column_queue = Queue()
+    for col in sorted(list(transform.keys())):
+        column_queue.put((col, False))
+
+    error_string = ""
+    while not column_queue.empty():
+        col, is_dep = column_queue.get()
+        if is_dep:
+            raise RuntimeError(
+                "Error: All remaining columns either have an error or cyclic dependencies:\n"
+                + error_string
+            )
+
+        column_info = transform[col]
+        is_created = False
+
+        if "sql" in column_info:
+            try:
```

Comment:
Is there a situation in which an exception here (or in the try/catch below) would be avoided later? If not, then why not simplify this logic and just error as soon as there is an exception?

Reply:
It's just non-cyclic dependencies because there is no assertions on the order the cols are processed. If col b relies on col a, if col b is tried be created first than it will raise a preventable exception and it will be avoided once col a is created.